tests/phpunit/tests/script-modules/wpScriptModules.php

```diff
@@ -732,4 +732,211 @@ public function test_wp_print_import_map_has_polyfill_when_modules_registered()
+			'Double quote'                           => array( '"', '\\"', 'UTF-8' ),
+			'Single quote'                           => array( '\'', '\'', 'UTF-8' ),
+			'Less than'                              => array( '<', '\u003C', 'UTF-8' ),
+			'Greater than'                           => array( '>', '\u003E', 'UTF-8' ),
+			'Ampersand'                              => array( '&', '&', 'UTF-8' ),
+			'Newline'                                => array( "\n", "\\n", 'UTF-8' ),
+			'Tab'                                    => array( "\t", "\\t", 'UTF-8' ),
+			'Form feed'                              => array( "\f", "\\f", 'UTF-8' ),
+			'Carriage return'                        => array( "\r", "\\r", 'UTF-8' ),
+			'Line separator'                         => array( "\u{2028}", "\u{2028}", 'UTF-8' ),
+			'Paragraph separator'                    => array( "\u{2029}", "\u{2029}", 'UTF-8' ),
+
+			/*
+			 * The following is the Flag of England emoji
+			 * PHP: "\u{1F3F4}\u{E0067}\u{E0062}\u{E0065}\u{E006E}\u{E0067}\u{E007F}"
+			 */
+			'Flag of england'                        => array( '🏴󠁧󠁢󠁥󠁮󠁧󠁿', '🏴󠁧󠁢󠁥󠁮󠁧󠁿', 'UTF-8' ),
+			'Malicious script closer'                => array( '</script>', '\u003C/script\u003E', 'UTF-8' ),
+			'Entity-encoded malicious script closer' => array( '&lt;/script&gt;', '&lt;/script&gt;', 'UTF-8' ),
+
+			// Non UTF-8
+			'Solidus'                                => array( '/', '/', 'iso-8859-1' ),
+			'Less than'                              => array( '<', '\u003C', 'iso-8859-1' ),
+			'Greater than'                           => array( '>', '\u003E', 'iso-8859-1' ),
+			'Ampersand'                              => array( '&', '&', 'iso-8859-1' ),
+			'Newline'                                => array( "\n", "\\n", 'iso-8859-1' ),
+			'Tab'                                    => array( "\t", "\\t", 'iso-8859-1' ),
+			'Form feed'                              => array( "\f", "\\f", 'iso-8859-1' ),
+			'Carriage return'                        => array( "\r", "\\r", 'iso-8859-1' ),
+			'Line separator'                         => array( "\u{2028}", "\u2028", 'iso-8859-1' ),
+			'Paragraph separator'                    => array( "\u{2029}", "\u2029", 'iso-8859-1' ),
+			/*
+			 * The following is the Flag of England emoji
+			 * PHP: "\u{1F3F4}\u{E0067}\u{E0062}\u{E0065}\u{E006E}\u{E0067}\u{E007F}"
+			 */
+			'Flag of england'                        => array( '🏴󠁧󠁢󠁥󠁮󠁧󠁿', "\ud83c\udff4\udb40\udc67\udb40\udc62\udb40\udc65\udb40\udc6e\udb40\udc67\udb40\udc7f", 'iso-8859-1' ),
+			'Malicious script closer'                => array( '</script>', '\u003C/script\u003E', 'iso-8859-1' ),
+			'Entity-encoded malicious script closer' => array( '&lt;/script&gt;', '&lt;/script&gt;', 'iso-8859-1' ),
+
+		);
+	}
+
+	/**
+	 * @ticket 60647
+	 *
+	 * @dataProvider data_invalid_script_module_data
+	 * @param mixed $data Data to return in filter.
+	 */
+	public function test_print_script_module_data_does_not_print_bad_data( $data ) {
```

Comment:
What is bad data?

Reply:
Non-array data, data that filters shouldn't return.